### PR TITLE
Added dynamic adding of obstacles and cars

### PR DESCRIPTION
**Pull request by BeeHiveJava**
Ball will be added later.

### CODE REVIEW

**Comment by TehMessiah** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ClientController.java#L72](https://github.com/PTS3-S34A/User-Interface/blob/faf0aa6cde7a353cdb414c96a48f8886467a4cde/Soccar [UI]/src/nl/soccar/ui/rmi/ClientController.java#L72): Refactor this method to throw at most one checked exception instead of: java.io.IOException, java.rmi.NotBoundException [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1160)
1. MINOR [BallSyncMessageHandler.java#L11](https://github.com/PTS3-S34A/User-Interface/blob/faf0aa6cde7a353cdb414c96a48f8886467a4cde/Soccar [UI]/src/nl/socnet/message/handler/BallSyncMessageHandler.java#L11): Document this public class. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [PlayerSyncMessageHandler.java#L16](https://github.com/PTS3-S34A/User-Interface/blob/faf0aa6cde7a353cdb414c96a48f8886467a4cde/Soccar [UI]/src/nl/socnet/message/handler/PlayerSyncMessageHandler.java#L16): Document this public class. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [UI]/src/nl/socnet/message/SpawnObstacleMessage.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    private final float x;
+    private final float y;
+    private final float angle;
+
+    private final float width;
+    private final float height;
+    private final ObstacleType type;
+
+    public SpawnObstacleMessage(float x, float y, float angle, float width, float height, ObstacleType type) {
```
Comment (ghost) [inline Soccar [UI]/src/nl/socnet/message/SpawnObstacleMessage.java:23]: MINOR Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [UI]/src/nl/socnet/message/handler/SpawnObstacleMessageHandler.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+/**
+ * @author PTS34A
+ */
+public final class SpawnObstacleMessageHandler extends MessageHandler<SpawnObstacleMessage> {
+
+    @Override
+    protected void handle(Connection connection, SpawnObstacleMessage message) throws Exception {
+        Object controller = Main.getInstance().getController();
+        if (controller instanceof GameViewFXMLController) {
+            Player player = ClientController.getInstance().getCurrentPlayer();
```
Comment (TehMessiah) [inline Soccar [UI]/src/nl/socnet/message/handler/SpawnObstacleMessageHandler.java]: MAJOR Remove this unused "player" local variable. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1481)
MAJOR Remove this useless assignment to local variable "player". [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1854)

File: Soccar [UI]/src/nl/soccar/ui/fx/drawable/MapUiFx.java [hunk 3]
```diff
@@ -29,8 +26,6 @@
     private static final float WALL_WIDTH;
```
Comment (ghost) [inline Soccar [UI]/src/nl/soccar/ui/fx/drawable/MapUiFx.java:26]: MAJOR Remove this unused "WALL_WIDTH" private field. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1068)

File: Soccar [UI]/src/nl/soccar/ui/fx/drawable/MapUiFx.java [hunk 4]
```diff
@@ -29,8 +26,6 @@
     private static final float WALL_WIDTH;
     private static final float CORNER_SIZE;
```
Comment (ghost) [inline Soccar [UI]/src/nl/soccar/ui/fx/drawable/MapUiFx.java:27]: MAJOR Remove this unused "CORNER_SIZE" private field. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1068)

File: Soccar [UI]/src/nl/socnet/message/PlayerChangedGameStatusMessage.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+/**
+ * @author PTS34A
+ */
+@MessageEvent(id = MessageConstants.PLAYER_CHANGED_GAME_STATUS_MESSAGE_ID, handler = PlayerChangedGameStatusMessageHandler.class)
+public final class PlayerChangedGameStatusMessage extends Message {
+
+    private final Status status;
+
+    public PlayerChangedGameStatusMessage(Status status) {
```
Comment (ghost) [inline Soccar [UI]/src/nl/socnet/message/PlayerChangedGameStatusMessage.java:16]: MINOR Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [UI]/src/nl/socnet/message/PlayerChangedGameStatusMessage.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    public Status getStatus() {
+        return status;
+    }
+
+    @Override
+    public int getId() {
+        return MessageConstants.PLAYER_CHANGED_GAME_STATUS_MESSAGE_ID;
+    }
+
+    public enum Status {
```
Comment (ghost) [inline Soccar [UI]/src/nl/socnet/message/PlayerChangedGameStatusMessage.java:29]: MINOR Document this public enum. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [UI]/src/nl/soccar/ui/fx/controller/GameViewFXMLController.java [hunk 7]
```diff
@@ -69,21 +59,51 @@ public void initialize(URL url, ResourceBundle rb) {
...
         initializeBall(session, gameCanvas);
-        initializeCars(session, gameCanvas);
         initializeNotification(session, gameCanvas);
 
-        GameEngine engine = gameCanvas.getGameEngine();
+        engine = gameCanvas.getGameEngine();
         engine.start();
+    }
+
+    public void spawnEntity(Entity entity) {
```
Comment (ghost) [inline Soccar [UI]/src/nl/soccar/ui/fx/controller/GameViewFXMLController.java:68]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [UI]/src/nl/socnet/message/ChangePlayerStatusMessage.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+/**
+ * @author PTS34A
+ */
+@MessageEvent(id = MessageConstants.CHANGE_PLAYER_STATUS_MESSAGE_ID, handler = ChangePlayerStatusMessageHandler.class)
+public final class ChangePlayerStatusMessage extends Message {
+
+    private final Status status;
+
+    public ChangePlayerStatusMessage(Status status) {
```
Comment (ghost) [inline Soccar [UI]/src/nl/socnet/message/ChangePlayerStatusMessage.java:16]: MINOR Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [UI]/src/nl/socnet/message/ChangePlayerStatusMessage.java [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+    public Status getStatus() {
+        return status;
+    }
+
+    @Override
+    public int getId() {
+        return MessageConstants.CHANGE_PLAYER_STATUS_MESSAGE_ID;
+    }
+
+    public enum Status {
```
Comment (ghost) [inline Soccar [UI]/src/nl/socnet/message/ChangePlayerStatusMessage.java:29]: MINOR Document this public enum. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [UI]/src/nl/socnet/message/SpawnCarMessage.java [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+ */
+@MessageEvent(id = MessageConstants.SPAWN_CAR_MESSAGE_ID, handler = SpawnCarMessageHandler.class)
+public final class SpawnCarMessage extends Message {
+
+    private final int playerId;
+    private final float x;
+    private final float y;
+    private final float angle;
+
+    public SpawnCarMessage(int playerId, float x, float y, float angle) {
```
Comment (ghost) [inline Soccar [UI]/src/nl/socnet/message/SpawnCarMessage.java:19]: MINOR Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

**Comment by TehMessiah** [comment]
SonarQube analysis reported 16 issues
* MAJOR 5 major
* MINOR 11 minor

Watch the comments in this conversation to review them.

#### 5 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ClientController.java#L72](https://github.com/PTS3-S34A/User-Interface/blob/c47f07b2c5ddaa877dfd7039b0074ad806138f13/Soccar [UI]/src/nl/soccar/ui/rmi/ClientController.java#L72): Refactor this method to throw at most one checked exception instead of: java.io.IOException, java.rmi.NotBoundException [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1160)
1. MINOR [GameCanvasFx.java#L18](https://github.com/PTS3-S34A/User-Interface/blob/c47f07b2c5ddaa877dfd7039b0074ad806138f13/Soccar [UI]/src/nl/soccar/ui/fx/GameCanvasFx.java#L18): Remove this duplicated import. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [GameCanvasFx.java#L19](https://github.com/PTS3-S34A/User-Interface/blob/c47f07b2c5ddaa877dfd7039b0074ad806138f13/Soccar [UI]/src/nl/soccar/ui/fx/GameCanvasFx.java#L19): Remove this duplicated import. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [BallSyncMessageHandler.java#L11](https://github.com/PTS3-S34A/User-Interface/blob/c47f07b2c5ddaa877dfd7039b0074ad806138f13/Soccar [UI]/src/nl/socnet/message/handler/BallSyncMessageHandler.java#L11): Document this public class. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [PlayerSyncMessageHandler.java#L16](https://github.com/PTS3-S34A/User-Interface/blob/c47f07b2c5ddaa877dfd7039b0074ad806138f13/Soccar [UI]/src/nl/socnet/message/handler/PlayerSyncMessageHandler.java#L16): Document this public class. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [UI]/src/nl/socnet/message/LeaveSessionMessage.java [hunk 11]
```diff
@@ -1,20 +1,20 @@
...
+import nl.soccar.socnet.message.Message;
+import nl.soccar.socnet.message.MessageConstants;
+import nl.soccar.socnet.message.MessageEvent;
+import nl.socnet.message.handler.LeaveSessionMessageHandler;
+
+/**
+ *
+ * @author PTS34A
+ */
+@MessageEvent(id = MessageConstants.PLAYER_LEAVE_SESSION_MESSAGE_ID, handler = LeaveSessionMessageHandler.class)
```
Comment (Ranray) [inline Soccar [UI]/src/nl/socnet/message/LeaveSessionMessage.java]: Graag de constants aanpassen naar LEAVE_SESSION_MESSAGE_ID

**Review by Ranray — CHANGES_REQUESTED** [review]

File: Soccar [UI]/src/nl/soccar/ui/input/InputController.java [hunk 12]
```diff
@@ -1,39 +1,41 @@
...
         throttleAction = ThrottleAction.IDLE;
         handbrakeAction = HandbrakeAction.INACTIVE;
         steerAction = SteerAction.NONE;
     }
 
+    public static InputController getInstance() {
+        return INSTANCE;
+    }
+
     public void initializeInput(Keyboard keyboard) {
```
Comment (ghost) [inline Soccar [UI]/src/nl/soccar/ui/input/InputController.java:39]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

**Comment by TehMessiah** [comment]
SonarQube analysis reported 19 issues
* MAJOR 7 major
* MINOR 12 minor

Watch the comments in this conversation to review them.

#### 7 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [CreateRoomFXMLController.java#L152](https://github.com/PTS3-S34A/User-Interface/blob/d72c96d8d69f91f9d7812491a93575f0e09ef4aa/Soccar [UI]/src/nl/soccar/ui/fx/controller/CreateRoomFXMLController.java#L152): Remove this unused method parameter "password". [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1172)
1. MAJOR [InputController.java#L60](https://github.com/PTS3-S34A/User-Interface/blob/d72c96d8d69f91f9d7812491a93575f0e09ef4aa/Soccar [UI]/src/nl/soccar/ui/input/InputController.java#L60): This branch's code block is the same as the block for the branch on line 54. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1871)
1. MAJOR [ClientController.java#L72](https://github.com/PTS3-S34A/User-Interface/blob/d72c96d8d69f91f9d7812491a93575f0e09ef4aa/Soccar [UI]/src/nl/soccar/ui/rmi/ClientController.java#L72): Refactor this method to throw at most one checked exception instead of: java.io.IOException, java.rmi.NotBoundException [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1160)
1. MINOR [GamePad.java#L33](https://github.com/PTS3-S34A/User-Interface/blob/d72c96d8d69f91f9d7812491a93575f0e09ef4aa/Soccar [UI]/src/nl/soccar/ui/input/GamePad.java#L33): Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [GamePad.java#L39](https://github.com/PTS3-S34A/User-Interface/blob/d72c96d8d69f91f9d7812491a93575f0e09ef4aa/Soccar [UI]/src/nl/soccar/ui/input/GamePad.java#L39): Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [BallSyncMessageHandler.java#L11](https://github.com/PTS3-S34A/User-Interface/blob/d72c96d8d69f91f9d7812491a93575f0e09ef4aa/Soccar [UI]/src/nl/socnet/message/handler/BallSyncMessageHandler.java#L11): Document this public class. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [PlayerSyncMessageHandler.java#L16](https://github.com/PTS3-S34A/User-Interface/blob/d72c96d8d69f91f9d7812491a93575f0e09ef4aa/Soccar [UI]/src/nl/socnet/message/handler/PlayerSyncMessageHandler.java#L16): Document this public class. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

**Review by Ranray — APPROVED** [review]

**Comment by ghost** [comment]
SonarQube analysis reported 17 issues
* MAJOR 5 major
* MINOR 12 minor

Watch the comments in this conversation to review them.

#### 7 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [CreateRoomFXMLController.java#L152](https://github.com/PTS3-S34A/User-Interface/blob/c89b7818bfa5a9ca0045ac19b611719ae0a7872d/Soccar [UI]/src/nl/soccar/ui/fx/controller/CreateRoomFXMLController.java#L152): Remove this unused method parameter "password". [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1172)
1. MAJOR [InputController.java#L60](https://github.com/PTS3-S34A/User-Interface/blob/c89b7818bfa5a9ca0045ac19b611719ae0a7872d/Soccar [UI]/src/nl/soccar/ui/input/InputController.java#L60): This branch's code block is the same as the block for the branch on line 54. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1871)
1. MAJOR [ClientController.java#L72](https://github.com/PTS3-S34A/User-Interface/blob/c89b7818bfa5a9ca0045ac19b611719ae0a7872d/Soccar [UI]/src/nl/soccar/ui/rmi/ClientController.java#L72): Refactor this method to throw at most one checked exception instead of: java.io.IOException, java.rmi.NotBoundException [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1160)
1. MINOR [GamePad.java#L33](https://github.com/PTS3-S34A/User-Interface/blob/c89b7818bfa5a9ca0045ac19b611719ae0a7872d/Soccar [UI]/src/nl/soccar/ui/input/GamePad.java#L33): Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [GamePad.java#L39](https://github.com/PTS3-S34A/User-Interface/blob/c89b7818bfa5a9ca0045ac19b611719ae0a7872d/Soccar [UI]/src/nl/soccar/ui/input/GamePad.java#L39): Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [BallSyncMessageHandler.java#L11](https://github.com/PTS3-S34A/User-Interface/blob/c89b7818bfa5a9ca0045ac19b611719ae0a7872d/Soccar [UI]/src/nl/socnet/message/handler/BallSyncMessageHandler.java#L11): Document this public class. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [PlayerSyncMessageHandler.java#L16](https://github.com/PTS3-S34A/User-Interface/blob/c89b7818bfa5a9ca0045ac19b611719ae0a7872d/Soccar [UI]/src/nl/socnet/message/handler/PlayerSyncMessageHandler.java#L16): Document this public class. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

**Pull request merged**